dts/arm/microchip/mec172x/mec172xnsz-pinctrl.dtsi

```diff
@@ -0,0 +1,966 @@
+/*
+ * Copyright (c) 2021 Microchip Technology Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <dt-bindings/pinctrl/mchp-xec-pinctrl.h>
+
+/ {
+	soc {
+		pinctrl: pin-controller@40081000 {
+
+			/* ADC */
+			pinctrl_adc0_in0: adc0_in0 {
+				pinmux = < MCHP_XEC_PINMUX(0200, AF1) >;
+			};
+
+			pinctrl_adc0_in1: adc0_in1 {
+				pinmux = < MCHP_XEC_PINMUX(0201, AF1) >;
+			};
+
+			pinctrl_adc0_in2: adc0_in2 {
+				pinmux = < MCHP_XEC_PINMUX(0202, AF1) >;
+			};
+
```

Comment:
note, non-blocking: consider using the grouping approach (is less verbose when hand-written). For example, you'd have:

```
&adc0_default {
    group1 {
        pinmux = <MCHP_XEC_PINMUX(0200, AF1)>,
                        <MCHP_XEC_PINMUX(0201, AF1)>,
                        <MCHP_XEC_PINMUX(0202, AF1)>,
                        <MCHP_XEC_PINMUX(0203, AF1)>,
                        <MCHP_XEC_PINMUX(0204, AF1)>,
                        <MCHP_XEC_PINMUX(0205, AF1)>,
                        <MCHP_XEC_PINMUX(0206, AF1)>,
                        <MCHP_XEC_PINMUX(0207, AF1)>,
                        <MCHP_XEC_PINMUX(067, AF1)>;
    };
};
```

and then

```
&adc0 {
    pinctrl-0 = <&adc0_default>;
    pinctrl-names = "default";
};
```

The current pinctrl file is ~1K lines, which seems too much really. The approach used here works better when autogeneration is done.

Reply:
another comment: please use a better name for nodes if going with current approach, e.g

```
adc0_in0_default
```

^- peripheral adc0, signal in0, default state

Reply:
We don't want to group the pins since we want to allow selective configuration of the pins in the supposed group. For example for adc, we will not be enabling all the adc channels. Most customers are not enabling all the adc channels all at once.

Also, other pinctrl implementations (of other vendors) are not using group feature.

Reply:
> We don't want to group the pins since we want to allow selective configuration of the pins in the supposed group. For example for adc, we will not be enabling all the adc channels. Most customers are not enabling all the adc channels all at once.

The same functionality can be achieved with groups, it's just about representation formats.

Reply:
Not considering grouping approach for now.